public/app/plugins/panel/state-timeline/StateTimelineTooltip.tsx

```diff
@@ -32,6 +32,17 @@ export const StateTimelineTooltip: React.FC<StateTimelineTooltipProps> = ({
   const xFieldFmt = xField.display || getDisplayProcessor({ field: xField, timeZone, theme });
 
   const field = alignedData.fields[seriesIdx!];
+
+  /**
+   * There could be a case when the tooltip shows a data from one of a multiple query and the other query finishes first
+   * from refreshing. This causes data to be out of sync. alignedData - 1 because Time field doesn't count.
+   * Render nothing in this case to prevent error.
+   * See https://github.com/grafana/support-escalations/issues/932
+   */
+  if (alignedData.fields.length - 1 !== data.length || !field) {
+    return null;
+  }
```

Comment:
Think this should be handled in https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/state-timeline/StateTimelinePanel.tsx#L38 to avoid a moment, when there is an empty tooltip rendered (as seen on the recording)

Reply:
yeah that make sense.